### PR TITLE
Enhancement/deleteanswer

### PR DESCRIPTION
**Pull request by mshashank0**
Delete answer based on role
Validations added
Exception handled

### CODE REVIEW

File: quora-service/src/main/java/com/upgrad/quora/service/business/AnswerBusinessService.java [hunk 1]
```diff
@@ -28,4 +32,39 @@ public AnswerEntity createAnswer(AnswerEntity answerEntity, final String authori
         answerEntity.setUser(userAuthToken.getUserEntity());
         return answerDao.createAnswer(answerEntity);
     }
+
+    @Transactional(propagation = Propagation.REQUIRED)
```
Comment (post2shyam) [inline quora-service/src/main/java/com/upgrad/quora/service/business/AnswerBusinessService.java:36]: Please remove this. Not required for Transactional.

File: quora-service/src/main/java/com/upgrad/quora/service/dao/AnswerDao.java [hunk 2]
```diff
@@ -19,11 +21,42 @@ public AnswerEntity createAnswer(AnswerEntity answerEntity) {
...
 
+    public AnswerEntity getAnswerByUuId(final String uuid) {
+        try {
+            return entityManager.createNamedQuery("AnswerFromUuid", AnswerEntity.class).setParameter("answerUuid", uuid).getSingleResult();
+        } catch (NoResultException nre) {
+            return null;
+        }
+    }
+
+    public Boolean isUserOwnerOfAnswer(String answerUuid, String userUuid) {
```
Comment (post2shyam) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/AnswerDao.java:32]: Please add final-keywords.

File: quora-service/src/main/java/com/upgrad/quora/service/dao/AnswerDao.java [hunk 3]
```diff
@@ -19,11 +21,42 @@ public AnswerEntity createAnswer(AnswerEntity answerEntity) {
...
+            else {
+                return true;
+            }
+        }catch (NoResultException nre)
+        {
+            return false;
+        }
+    }
+
+    public AnswerEntity deleteAnswer(AnswerEntity answerEntity)
```
Comment (post2shyam) [inline quora-service/src/main/java/com/upgrad/quora/service/dao/AnswerDao.java:48]: This operation is transactional. Please introduce.

File: quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java [hunk 4]
```diff
@@ -9,9 +9,8 @@
 @Table(name = "answer" , schema = "public")
 @NamedQueries(
         {
-                @NamedQuery(name = "getAnswerFromId" , query = "select q from AnswerEntity q where q.uuid = :uuid"),
-                @NamedQuery(name = "checkAnswerBelongToUser" , query = "select a from AnswerEntity a INNER JOIN UserEntity u on a.user = u.id where a.uuid =:auuid and u.uuid = :uuuid"),
-                @NamedQuery(name = "getAllAnswers" , query = "select a from AnswerEntity a INNER JOIN QuestionEntity q on a.question = q.id where q.uuid = :uuid")
+                @NamedQuery(name = "AnswerFromUuid" , query = "select q from AnswerEntity q where q.uuid = :answerUuid"),
```
Comment (post2shyam) [inline quora-service/src/main/java/com/upgrad/quora/service/entity/AnswerEntity.java:12]: camelCase for query-name.

**Review by post2shyam — APPROVED** [review]
Please submit these review changes in the next pr. 
Approving this.

File: quora-service/src/main/java/com/upgrad/quora/service/dao/AnswerDao.java (hunk 3 above)
Comment (mshashank0) [reply quora-service/src/main/java/com/upgrad/quora/service/dao/AnswerDao.java:48]: Added in service class

**Pull request merged**